### DISCUSSION TITLE
PorousFlow: Hydromechanical Iterative Coupling Question

### DISCUSSION THREAD

**Discussion by jrpatterson79** [idaholab/moose#24367] [Q&A Modules: General] [answered]
Hello friends,

First time poster and new user to the MOOSE framework. I'm working to build a coupled hydromechanical discrete fracture model. My starting point is a 2D penny-shaped, horizontal, fracture embedded in a 3D rock mesh. I would like to use an iterative coupling scheme to couple the hydraulic problem to the quasi-static linear elastic mechanical problem using Multi-Apps and Transfers. 

I can run the model, in its current form, to completion without Picard (Fixed Point) iterations; however, when I add the Fixed Point iterations to the Executioner block I get the error below. I read this error as saying my injection location (a Dirac kernel) has the same node ID as one of the nodes in the 3D mesh, but I'm not sure how to correct it. Does anyone have any insights they can share?
```
*** ERROR ***
The following error occurred in the object "pp_from_frac", of type "ReporterPointSource".

Cached Dirac point (x,y,z)=(       0,        0,        0) already exists with ID: 0 and does not match point (x,y,z)=( 9.69032,  92.1973,        0)
```

The input files are located at [https://github.com/jrpatterson79/testing](https://github.com/jrpatterson79/testing) 
I updated MOOSE and all packages in the mamba moose environment this morning to ensure I'm running with the most recent release and package versions. 

Thanks in advance,
J Patt

**Comment by cpgr** [comment]
I'm not sure if I can help with this problem but just wanted to say this looks really cool! Hopefully it is something simple that one of the MOOSE team can help with.

**Reply by GiudGiud** [reply]
let me take a look today

**Reply by GiudGiud** [reply]
Ok I found the issue, the cache is not cleared and the reporter values (giving the points) change. 

I dont think this was in whoever designed Dirac Kernel's objectives to have moving points. It might not be too difficult to fix though

**Reply by GiudGiud** [reply, marked as answer]
For right now you can either check out my PR
https://github.com/idaholab/moose/pull/24400
or work with this object
https://mooseframework.inl.gov/source/dirackernels/ReporterTimePointSource.html

**Reply by jrpatterson79** [reply]
@GiudGiud, thank you for making the time to have a look and offer your insight. I will see what I can do with the ReporpterTimePointSource while the PR is being reviewed for merge.
